src/ResourceStateBuilder.js

```diff
@@ -0,0 +1,104 @@
+// Copyright 2019 Stanford University see LICENSE for license
+
+import shortid from 'shortid'
+
+const rdf = require('rdf-ext')
+const _ = require('lodash')
+
+/**
+ * Builds Redux state from an RDF graph
+ */
+export default class ResourceStateBuilder {
+  /**
+   * @param {Dataset} dataset the RDF graph
+   * @param {string} resourceURI URI of the resource
+   */
+  constructor(dataset, resourceURI) {
+    this.dataset = dataset
+    this.resourceURI = resourceURI
+    this.resourceState = {}
+  }
+
+  /**
+   * @return {Object} the resource represented as Redux state
+   */
+  get state() {
+    this.resourceState = this.buildResource(rdf.namedNode(this.resourceURI))
+
+    return this.resourceState
+  }
+
+  /**
+   * @param {rdf.Term} resourceTerm NamedNode or BlankNode of resource to be built
+   * @return {Object} the resource represented as Redux state
+   */
+  buildResource(resourceTerm) {
+    const thisResourceState = {}
+    // Find the resource template id. Should be only 1.
```

Comment:
some of these comments don't pass this Sandi Metz test:  if you read the code line out loud, and it says what your comment says, the comment is superfluous.   I think this could be like below `// expect exactly 1 rtid` and you can get rid of `//find the properties` comment below and `// add each property`